src/common/requireAuth.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import * as React from 'react';
+import { withNavigation } from 'react-navigation';
+import { inject, observer } from 'mobx-react';
+
+// todo work only on app intializetion, after that tabNavigator don't rerender
+// screens so componentwillMount isnn't invoked
+const requireAuth = BaseComponent =>
```

Comment:
I hoped this HOC would work like in brewsky.admin, but it doesn't. Basically, because TabNavigator(and I think most of other navigation types) mount component only once, so componentWillMount() won't be invoked every time and we can't rely on it.... But its a mobile app,  navigation should be very straightforward, so maybe we don't need this kind of stuff.